src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1069,6 +1075,46 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
             }
         }
 
+        internal static string GetDefaultHostId(ScriptSettingsManager settingsManager, ScriptHostConfiguration scriptConfig)
+        {
+            // We're setting the default here on the newly created configuration
+            // If the user has explicitly set the HostID via host.json, it will overwrite
+            // what we set here
+            string hostId = null;
+            if (scriptConfig.IsSelfHost)
+            {
+                // When running locally, derive a stable host ID from machine name
+                // and root path. We use a hash rather than the path itself to ensure
+                // IDs differ (due to truncation) between folders that may share the same
+                // root path prefix.
+                // Note that such an ID won't work in distributed scenarios, so should
+                // only be used for local/CLI scenarios.
+                var sanitizedPath = scriptConfig.RootScriptPath
```

Comment:
This is based on code from @ahmelsayed's https://github.com/Azure/azure-webjobs-sdk-script/pull/916, with some tweaks.